modules/server/server-status/src/main/java/com/enonic/xp/server/impl/status/ProbeServlet.java

```diff
@@ -0,0 +1,71 @@
+package com.enonic.xp.server.impl.status;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Activate;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.net.MediaType;
+
+import com.enonic.xp.server.impl.status.check.StateCheck;
+
+public abstract class ProbeServlet
+    extends HttpServlet
+{
+    private final StateCheck stateCheck;
+
+    @Activate
+    public ProbeServlet( final StateCheck stateCheck )
+    {
+        this.stateCheck = stateCheck;
+    }
+
+    abstract String getSuccessMessage();
+
+    abstract String getFailedMessage( final List<String> errorMessages );
+
+    @Override
+    protected void doGet( final HttpServletRequest req, final HttpServletResponse res )
+        throws IOException
+    {
+        final List<String> errorMessages = stateCheck.check().getErrorMessages();
+        serializeJson( res, errorMessages );
+    }
+
+    private void serializeJson( final HttpServletResponse res, final List<String> errorMessages )
+        throws IOException
+    {
+
+        final ObjectNode json = JsonNodeFactory.instance.objectNode();
+        String message;
+        if ( errorMessages.isEmpty() )
+        {
+            res.setStatus( 200 );
+            message = getSuccessMessage();
+        }
+        else
+        {
+            res.setStatus( 500 );
+            message = getFailedMessage( errorMessages );
```

Comment:
Not sure what is the benefit to mix 
- inheritance for getSuccessMessage, getFailedMessage
- policy, transferred via constructor
- getFailedMessage list of errors via arguments